internal/deepr/session_manager.go

```diff
@@ -51,13 +51,41 @@ func (sm *SessionManager) GetSession(userID, chatID string) (*ActiveSession, boo
 	return session, exists
 }
 
-// CreateSession creates a new active session
+// CreateSession creates a new active session.
 func (sm *SessionManager) CreateSession(userID, chatID string, backendConn *websocket.Conn, ctx context.Context, cancel context.CancelFunc) *ActiveSession {
 	sm.mu.Lock()
 	defer sm.mu.Unlock()
 
 	key := sm.getSessionKey(userID, chatID)
 
+	// Check if session already exists
+	if existingSession, exists := sm.sessions[key]; exists {
+		// Safely read client count
+		existingSession.mu.RLock()
+		existingClientCount := len(existingSession.clientConns)
+		existingSession.mu.RUnlock()
+
+		sm.logger.WithComponent("deepr-session").Warn("OVERWRITING existing session",
+			slog.String("user_id", userID),
+			slog.String("chat_id", chatID),
+			slog.String("session_key", key),
+			slog.Int("existing_client_count", existingClientCount))
+
+		// Proactively cancel and close sockets to avoid leaks
+		if existingSession.CancelFunc != nil {
+			existingSession.CancelFunc()
+		}
+		if existingSession.BackendConn != nil {
+			_ = existingSession.BackendConn.Close()
```

Comment:
If this is stored in memory, then it might break since there's multiple instances of proxy running